README.md

```diff
@@ -26,55 +25,65 @@ We use [ISA-JSON](https://isatools.readthedocs.io/en/latest/isamodel.html) to st
 
-6. **Registering ISA-JSON at the target repositories**: Sending the filtered ISA-JSON to the endpoints of the repositories who accept ISA-JSON, for example ENA.
+5. **Submitting Data to Target Repositories**  
+MARS-CLI uploads data using FTP. Some repositories require that data files are present in their upload space before metadata submission. This step ensures that data availability and checksum validation are completed before sending metadata.  
 
-    > The source code for the ISA-JSON API for ENA can be found in the [repository-services repo](/repository-services/isajson-ena/) and can be used for testing
+6. **Registering ISA-JSON at the Target Repositories**  
+The filtered ISA-JSON is submitted to repositories that accept ISA-JSON metadata, such as ENA. The MARS project helps out with the adaptation of ISA-JSON by the repositories using the so called adapters. 
 
-7. **Processing the receipts and errors from the repositories**: After a successful submission, each repository sends back a receipt in a standard format defined for MARS (see [repository-api info](/repository-services/repository-api.md)). The receipt contains the path of the objects in the ISA-JSON for which an accession number has been generated, and the related accession number. 
+- Source Code: The ISA-JSON API for ENA is available in the [repository-services repo](/repository-services/isajson-ena/) and can be used for testing.  
 
-8. **Updating the BioSamples External References**: Data broker uses the BioSamples accession numbers to download the submitted BioSamples JSON and extend the `External References` schema by adding the accession numbers provided by the other target archives.
+7. **Processing Repository Receipts and Errors**  
+After submission, each repository returns a receipt in a standardized format defined for MARS (see [repository-api info](/repository-services/repository-api.md)).  
 
-9. **Dumping back an updated ISA-JSON with repositories' information**: Based on the information in the receipts, the ISA-JSON is populated with accession numbers linked to the submitted metadata objects and can be given back as output.
+- The receipt includes the paths of objects within the ISA-JSON and their assigned accession numbers.  
+- Errors encountered during submission are processed and reported accordingly.  
 
+8. **Updating BioSamples External References**  
+The Data Broker retrieves the BioSamples JSONs of the submitted samples using its accession numbers and updates the `External References` schema by adding the accession numbers assigned by other target repositories.  
+
+9. **Generating an Updated ISA-JSON with Repository Information**  
+Based on the information in repository receipts, the ISA-JSON is updated with accession numbers linked to submitted metadata objects. This final, enriched ISA-JSON can then be provided as output.  
 
 #### Credential management
 
```

Comment:
Seeing as we did add some basic credential management to MARS-CLI, we should probably change this section...